### PR TITLE
add git-sync

### PR DESCRIPTION
**Pull request by kumon**
This command syncs local branch with remote branch.

This runs:
```
git fetch <remote-name> <branch-name>
git reset --hard <remote-name>/<branch-name>
git clean -d -f -x
```

### CODE REVIEW

File: bin/git-sync [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+Example:
+  ${command} origin master
+EOS
+}
+
+function main()
+{
+  while [ "$1" != "" ]; do
+    case $1 in
+      -h | --help | help | "?" )
```
Comment (spacewander) [inline bin/git-sync]: `help` can be a legal name of a remote branch.

Comment (kumon) [inline bin/git-sync]: Thank you for reviewing it. I should fix that.

**Comment by spacewander** [comment]
What about using a default remote/branch when `remote/branch` argument is missing?
Just follow the convention of `git fetch/pull`.

**Comment by kumon** [comment]
I will follow the convention of `git fetch/pull`. It's more practical. Thank you so much.

File: bin/git-sync (hunk 1 above)
Comment (nicolaiskogheim) [inline bin/git-sync]: As we don't use "?" for seeing usage with any other commands, we shouldn't do it here either.
`?` is a special character in some terminals with `extglob` (i think) enabled, and could confuse people as well.

Comment (kumon) [inline bin/git-sync]: I will remove '?'. Thank you!

**Comment by kumon** [comment]
I changed to use upstream of local branch by default, when remote/branch is not specified.

File: bin/git-sync [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        fi
+        ;;
+    esac
+    shift
+  done
+
+  if [ "${remote}" = "" ]; then
+    local remote_branch="$(git rev-parse --abbrev-ref --symbolic-full-name @{u})"
+    branch="$(git rev-parse --abbrev-ref --symbolic-full-name @)"
+    remote="${remote_branch%%/${branch}}"
```
Comment (spacewander) [inline bin/git-sync]: A little quibble:
`git rev-parse --abbrev-ref --symbolic-full-name @{u}` will return nothing if there is not related upstream branch. 
Then users will see `... & sync with ''? [y/N]` and the command may simply reset to current branch, which is not really wanted.

**Comment by kumon** [comment]
I added upstream check that is based on exit status of `git rev-parse --abbrev-ref --symbolic-full-name @{u}`.

**Comment by spacewander** [comment]
@hemanth  @qw3rtman 
It looks good to me. What about your idea?

**Comment by qw3rtman** [comment]
Yup, looks good to me! Merging.

In the future, I think it'd be a good idea to standardize how help/usage is displayed for each command. Each command should have the same name for the function that displays the help/usage. Additionally, the formatting of the usage should be the same.

**Comment by kumon** [comment]
Thank you for reviewing it!

**Comment by qw3rtman** [comment]
Thanks for contributing, @kumon! :smile:

**Pull request merged**